app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -259,6 +260,11 @@ function FormSteps({
     didSubmitWithErrors.current = false;
   }, [activeErrors]);
 
+  // reset stepName if it doesn't correspond to an existing step
+  if (stepName && !steps.map((step) => step.name).includes(stepName)) {
+    setStepName(undefined);
+  }
```

Comment:
We should avoid side effects in the render logic of the component ([ref](https://beta.reactjs.org/learn/keeping-components-pure#side-effects-unintended-consequences)). We could probably move this inside a `useEffect`, e.g.

```suggestion
  useEffect(() => {
    // reset stepName if it doesn't correspond to an existing step
    if (stepName && !steps.map((step) => step.name).includes(stepName)) {
      setStepName(undefined);
    }
  }, [stepName, steps]);
```

Reply:
good catch, thank you! fixed in 73179dccff23ebe0e67c904be625fc4da5546ac8